src/config/session.c

```diff
@@ -176,7 +252,15 @@ session_environment_init(void)
 
 	for (struct wl_list *elm = iter(&paths); elm != &paths; elm = iter(elm)) {
 		struct path *path = wl_container_of(elm, path, link);
+
+		/* Process an environment file itself */
 		bool success = read_environment_file(path->string);
+
+		/* Process a correponding environment.d directory */
+		if (read_environment_dir(path->string)) {
+			success = true;
+		}
```

Comment:
Nitpick only
```c
success |= read_environment_dir(path->string);
```

Reply:
I don't think this is done anywhere else for Boolean overrides, and I usually like to avoid using bitwise operators when treating Booleans as opaque types.

Reply:
We do that for various `consumed_by_frame_context |=` operations in `src/input/cursor.c` but again, just a nitpick. Totally fine with keeping the current variant.

Reply:
Somehow I missed those cases. Consistency seems preferable here, so I included the suggestion.